pkg/tsdb/kairosdb/kairosdb.go

```diff
@@ -0,0 +1,186 @@
+package kairosdb
+
+import (
+	"context"
+	"fmt"
+	"path"
+	"strings"
+
+	"golang.org/x/net/context/ctxhttp"
+
+	"encoding/json"
+	"io/ioutil"
+	"net/http"
+	"net/url"
+
+	"github.com/grafana/grafana/pkg/components/null"
+	"github.com/grafana/grafana/pkg/log"
+	"github.com/grafana/grafana/pkg/models"
+	"github.com/grafana/grafana/pkg/setting"
+	"github.com/grafana/grafana/pkg/tsdb"
+	"github.com/grafana/grafana/pkg/tsdb/kairosdb/response"
+)
+
+type KairosdbExecutor struct {
+	*models.DataSource
+	httpClient *http.Client
+}
+
+func NewKairosdbExecutor(datasource *models.DataSource) (tsdb.Executor, error) {
+	plog.Info("Inside NewKairosdbExecutor - 0")
+	httpClient, err := datasource.GetHttpClient()
+
+	plog.Info("Inside NewKairosdbExecutor - 1")
+
+	if err != nil {
+		return nil, err
+	}
+	plog.Info("Inside NewKairosdbExecutor - 2")
+	return &KairosdbExecutor{
+		DataSource: datasource,
+		httpClient: httpClient,
+	}, nil
+}
+
+var (
+	plog log.Logger
+)
+
+func init() {
+	plog = log.New("tsdb.kairosdb1")
+	var exec = NewKairosdbExecutor
+	plog.Info("Exec value is ", exec)
+	tsdb.RegisterExecutor("grafana-kairosdb-datasource", exec)
+}
+
+func (e *KairosdbExecutor) Execute(ctx context.Context, queries tsdb.QuerySlice, queryContext *tsdb.QueryContext) *tsdb.BatchResult {
+
+	result := &tsdb.BatchResult{}
+
+	kairosdbQuery := NewQueryBuilder()
+	kairosdbQuery.SetAbsoluteStart(queryContext.TimeRange.MustGetFrom().UTC())
+	kairosdbQuery.SetAbsoluteEnd(queryContext.TimeRange.MustGetTo().UTC())
+	plog.Debug("Start_absolute", kairosdbQuery.AbsoluteStart)
+	plog.Debug("end_absolute", kairosdbQuery.AbsoluteEnd)
+
+	for _, query := range queries {
+		metric := e.buildMetric(query)
+		kairosdbQuery.AddMetric(metric)
+	}
+
+	if setting.Env == setting.DEV {
+		plog.Debug("KairosDb request", "params", kairosdbQuery)
+	}
+
+	req, err := e.createRequest(kairosdbQuery)
+	if err != nil {
+		result.Error = err
+		return result
+	}
+
+	res, err := ctxhttp.Do(ctx, e.httpClient, req)
+	if err != nil {
+		result.Error = err
+		return result
+	}
+
+	queryResult, err := e.parseResponse(res)
+	if err != nil {
+		return result.WithError(err)
+	}
+
+	result.QueryResults = queryResult
+	return result
+}
+
+func (e *KairosdbExecutor) createRequest(kairosdbQuery QueryBuilder) (*http.Request, error) {
+	u, _ := url.Parse(e.Url)
+	u.Path = path.Join(u.Path, "api/v1/datapoints/query/")
```

Comment:
In that case I will create seperate story to put this in config.ini file. Not sure how to add it in ini file.